Content.Server/Atmos/Rotting/RottingSystem.cs

```diff
@@ -98,6 +99,13 @@ public bool IsRotProgressing(EntityUid uid, PerishableComponent? perishable)
             return false;
         }
 
+        if (TryComp<BuckleComponent>(uid, out var buckle) &&
+            buckle is { Buckled: true, BuckledTo: not null } &&
+            HasComp<ActiveAntiRottingOnBuckleComponent>(buckle.BuckledTo.Value))
+        {
+            return false;
+        }
+
```

Comment:
Use the rottingevent.

Reply:
I updated the PR to use the event. Just had to change it slightly so that `Handled = true` prevents rotting instead of allowing it.